ui/app/AppLayouts/WalletV2/components/AddAccountPopup.qml

```diff
@@ -0,0 +1,210 @@
+import QtQuick 2.13
+import QtQuick.Controls 2.13
+import QtQuick.Dialogs 1.3
+import "../../../../imports"
+import "../../../../shared"
+import "../../../../shared/status"
+
+ModalPopup {
+    id: popup
+    width: 574
+    height: 498
+    title: qsTr("Add account")
+
+    property int marginBetweenInputs: 20
+    property string passwordValidationError: ""
+    property string accountNameValidationError: ""
+    property string keyOrSeedInputError: ""
+    property bool loading: false
+    property bool seedPhraseInserted: keyOrSeedPhraseInput.text.includes(" ")
+    property var onAfterAddAccount: function() {}
+
+    function reset() {
+        passwordInput.text = "";
+        accountNameInput.text = "";
+        keyOrSeedPhraseInput.text = "";
+        keyOrSeedInputError = "";
+        passwordValidationError = "";
+        accountNameValidationError = "";
+    }
+
+    function validate() {
+        if (passwordInput.text === "") {
+            passwordValidationError = qsTrId("you-need-to-enter-a-password");
+        } else if (passwordInput.text.length < 6) {
+            passwordValidationError = qsTrId("password-needs-to-be-6-characters-or-more");
+        } else {
+            passwordValidationError = "";
+        }
+
+        if (accountNameInput.text === "") {
+            accountNameValidationError = qsTrId("you-need-to-enter-an-account-name");
+        } else {
+            accountNameValidationError = "";
+        }
+
+        if (!seedPhraseInserted) {
+            if (keyOrSeedPhraseInput.text === "") {
+                keyOrSeedInputError = qsTr("You need to enter a valid private key or seed phrase");
+            } else {
+                if (!seedPhraseInserted) {
+                    keyOrSeedInputError = !Utils.isPrivateKey(keyOrSeedPhraseInput.text) ?
+                                qsTrId("enter-a-valid-private-key-(64-characters-hexadecimal-string)") : "";
+                } else {
+                    keyOrSeedInputError = !Utils.isMnemonic(keyOrSeedPhraseInput.text) ? qsTrId("enter-a-valid-mnemonic") : "";
+                }
+            }
+        }
+
+        return ((passwordValidationError === "") && (keyOrSeedInputError === "")
+                && (accountNameValidationError === ""));
+    }
+
+    function validateSeed() {
+        keyOrSeedInputError = onboardingModel.validateMnemonic(keyOrSeedPhraseInput.text);
+        const regex = new RegExp('word [a-z]+ not found in the dictionary', 'i');
+        if (regex.test(keyOrSeedInputError)) {
+            keyOrSeedInputError = qsTrId("custom-seed-phrase") + '. ' +
+                    qsTrId("custom-seed-phrase-text-1");
+        }
+        return (keyOrSeedInputError === "");
+    }
+
+    onOpened: {
+        reset();
```

Comment:
Instead of using a `reset` function, I've found it to be easier to destroy the popup on close:
```js
onClosed: {
  destroy()
}
```

Reply:
destroy() won't work here, since there is no one to create it again and I am not sure what is more performant in this case, dynamic creation/deletion or just keep the component alive - since we have no bindings that could revaluate all the time it shouldn't be an issue keeping the popup alive.

Reply:
Possibly we should look at dynamically creating the popup instead of keeping it alive in memory? This is how we have done for most popups in other parts of the codebase. Take a look at `openPopup` in `AppMain.qml`. Keeping track of popup state is also less clean and introduces bugs in my opinion, and creating/destroying the popup is more memory efficient.

Reply:
Done. However this should be improved, in general. We should think about having a popup loader in a single place rather than instantiating one in each file where from a popup should open

Reply:
Is the `openPopup` method (mentioned above) not suitable as a generalised option here?

Reply:
In order to use that function inside AddAccount.qml I would have to apply dynamic scoping which is not a good practice (see: https://doc.qt.io/qt-5/qtqml-documents-scope.html "_Dynamic scoping is very powerful, but it must be used cautiously to prevent the behavior of QML code from becoming difficult to predict. In general it should only be used in cases where the two components are already tightly coupled in another way. When building reusable components, it is preferable to use property interfaces, like this..._") so I would rather leave it as is now and go for a proper solution that will apply to all popups later on as part of refactoring. BTW, all popups in AddAccount.qml in Wallet 1 are kept always alive, was that on purpose?